test/functionalTest/cases/3121_missing_or_empty_metadata_values/entitiy_creation.test

```diff
@@ -0,0 +1,408 @@
+echo
+echo
+
+
+echo "02. GET /v2/entities/Room1."
+echo "==========================="
+orionCurl --url /v2/entities/Room1
+echo
+echo
+
+
+echo "03. POST /v2/entities/Room1/attrs, to add metadata with "" value to temperature and add pressure attribute with metadata with no value."
+echo "======================================================================================================================================="
+payload='{
+  "temperature": {
+    "value": 25,
+    "type": "Float",
+	"metadata": {
+      "accuracy": {
+        "value": "",
+        "type": "Float"
+      }
+    }
+  },
+  "pressure": {
+    "value": 720,
+    "type": "Integer",
+	"metadata": {
+      "accuracy": {
+        "type": "Float"
+      }
+    }
+  }
+}'
+orionCurl --url /v2/entities/Room1/attrs --payload "$payload"
+echo
+echo
+
+
+echo "04. GET /v2/entities/Room1."
+echo "==========================="
+orionCurl --url /v2/entities/Room1
+echo
+echo
+
+
+echo "05. POST /v2/entities/Room1/attrs, to update temperature with metadata with no value, pressure with metadata with regular value and add humidity attribute with metadata with "" value."
+echo "=========================================================================================================================================================="
+payload='{
+  "temperature": {
+    "value": 25,
+    "type": "Float",
+        "metadata": {
+      "accuracy": {
+        "type": "Float"
+      }
+    }
+  },
+  "pressure": {
+    "value": 720,
+    "type": "Integer",
+        "metadata": {
+      "accuracy": {
+	  	"value": "12.3",
+        "type": "Float"
+      }
+    }
+  },
+  "humidity": {
+    "value": 120,
+    "type": "Integer",
+        "metadata": {
+      "accuracy": {
+      	"value": "",
+        "type": "Float"
+      }
+    }
+  }
+}'
+orionCurl --url /v2/entities/Room1/attrs --payload "$payload"
+echo
+echo
+
+
+echo "06. GET /v2/entities/Room1."
+echo "==========================="
+orionCurl --url /v2/entities/Room1
+echo
+echo
+
+
+echo "07. POST /v2/entities, to create Room2 with temperature having two metadata one with "" value and another without value."
+echo "========================================================================================================================"
+payload='{
+  "id": "Room2",
+  "type": "Room",
+  "temperature": {
+    "value": 26.5,
+    "type": "Float",
+    "metadata": {
+      "accuracy": {
+        "value": "",
+        "type": "Float"
+      },
+     "tolerance": {
+        "type": "Float"
+      }
+    }
+  }
+}'
+orionCurl --url /v2/entities --payload "$payload"
+echo
+echo
+
+
+echo "08. GET /v2/entities/Room2."
+echo "==========================="
+orionCurl --url /v2/entities/Room2
+echo
+echo
+
+
+echo "09. PUT /v2/entities/Room2/attrs/temperature/value, updating the value of Room2 temperature attribute."
+echo "======================================================================================================"
+payload='{
```

Comment:
Note the accuracy and tolerance metadata values are the same used in step 7. They should be different (e.g. the oposite, i.e. accuracy with missing value and tolerance with "" value) in order to check that step 9 is actually working properly.

Reply:
fixed in bcf8353